server/Sample/sfreerdp.c

```diff
@@ -823,15 +925,23 @@ static DWORD WINAPI test_peer_mainloop(LPVOID arg)
 	client->update->RefreshRect = tf_peer_refresh_rect;
 	client->update->SuppressOutput = tf_peer_suppress_output;
 	client->settings->MultifragMaxRequestSize = 0xFFFFFF; /* FIXME */
+
+	WINPR_ASSERT(client->Initialize);
 	client->Initialize(client);
+
 	context = (testPeerContext*)client->context;
+	WINPR_ASSERT(context);
+
 	WLog_INFO(TAG, "We've got a client %s", client->local ? "(local)" : client->hostname);
 
 	while (error == CHANNEL_RC_OK)
 	{
 		count = 0;
 		{
-			DWORD tmp = client->GetEventHandles(client, &handles[count], 32 - count);
+			DWORD tmp;
+
+			WINPR_ASSERT(client->GetEventHandles);
```

Comment:
assert for `GetEventHandles` and `CheckFileDescriptor` in every loop iteration doesn't make sense - assert before the loop is enough (same in `test_server_mainloop`)

Reply:
I think since internals might change function pointers due to bugs at any time it is (at least) for the sample server not a bad idea to always check.